src/openfermion/transforms/_binary_code_transform.py

```diff
@@ -0,0 +1,178 @@
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+""" The transform function that does Fermion-qubit mappings
+    based on a BinaryCode (arXiv:1712.07067) """
+
+import numpy
+
+from openfermion.ops import (BinaryCode,
+                             FermionOperator,
+                             QubitOperator,
+                             SymbolicBinary)
+
+
+def extractor(binary_op):
+    """ Applies the extraction superoperator to a binary expression
+     to obtain the corresponding qubit operators.
+
+    Args:
+        binary_op (SymbolicBinary): the binary term
+
+    Returns (QubitOperator): the qubit operator corresponding to the
+        binary terms
+    """
+    return_fn = 1
+    for term in binary_op.terms:
+        multiplier = 1
+        if len(term) == 1:
+            term = term[0]
+            if isinstance(term, (numpy.int32, numpy.int64, int)):
+                multiplier = QubitOperator('Z' + str(term))
+            else:
+                multiplier = -1
+        elif len(term) > 1:
+            multiplier = dissolve(term)
+
+        return_fn *= multiplier
+    return return_fn
+
+
+def dissolve(term):
+    """Decomposition helper. Takes a product of binary variables
+    and outputs the Pauli-string sum that corresponds to the
+    decomposed multi-qubit operator.
+
+    Args:
+        term (tuple): product of binary variables, i.e.: 'w0 w2 w3'
+
+    Returns (QubitOperator): superposition of Pauli-strings
+
+    Raises:
+        ValueError: if the _action is not 'W'
```

Comment:
I update docs. I think this was left over from prior implementation.

Reply:
hmm I remember fixing that - meaning I may have messed something up with my merging. I should check to see whats happening before you approve.

Reply:
oh nvm - I totally missed that one and fixed the same error within the raise.